libcontainer/process_linux.go

```diff
@@ -255,6 +255,19 @@ func (p *initProcess) execSetns() error {
 	return nil
 }
 
+// parentError attempts to read any error from the parent so we have better context
+// about the cause of the failure. If no error is found then a new system error for
+// err and cause is returned.
+func parentError(pipe *os.File, err error, cause string) error {
+	if ierr := parseSync(pipe, func(sync *syncT) error { return nil }); ierr != nil {
```

Comment:
@stevenh Can you please give an example of how you've tested that this improves the current error situation. I assumed that this patch would show it, but it appears that this error is handled properly by `master`. As far as I can tell this code is just prioritising `procError` over internal `libcontainer.Process` errors -- I'm not sure whether that makes sense.

```diff
diff --git a/libcontainer/factory_linux.go b/libcontainer/factory_linux.go
index 7f044bd45d5a..f88d8c25c719 100644
--- a/libcontainer/factory_linux.go
+++ b/libcontainer/factory_linux.go
@@ -275,6 +275,7 @@ func (l *LinuxFactory) StartInitialization() (err error) {
        }

        // If Init succeeds, syscall.Exec will not return, hence none of the defers will be called.
+       return fmt.Errorf("LOLWHAT")
        return i.Init()
 }
```

Output in both branches:

```
% sudo ./runc run ctr
container_linux.go:259: starting container process caused "process_linux.go:385: container init caused \"LOLWHAT\""
```